[Bug] Double promotion warning treated as error in gz_msgs

### Describe the bug

```
❯ make px4_sitl
-- PX4_GIT_TAG: v1.17.0-alpha1-656-gfd7edaa4fe
CMake Warning (dev) at CMakeLists.txt:176 (find_package):
  Policy CMP0148 is not set: The FindPythonInterp and FindPythonLibs modules
  are removed.  Run "cmake --help-policy CMP0148" for policy details.  Use
  the cmake_policy command to set the policy and suppress this warning.

This warning is for project developers.  Use -Wno-dev to suppress it.

-- Found PythonInterp: /Users/ermarinellone/Code/PX4-Autopilot/.venv/bin/python3 (found suitable version "3.14.3", minimum required is "3")
-- PX4 config file: /Users/ermarinellone/Code/PX4-Autopilot/boards/px4/sitl/default.px4board
-- PLATFORM posix
-- ROMFSROOT px4fmu_common
-- TESTING y
-- ETHERNET y
-- ROOT_PATH .
-- PARAM_FILE /fs/mtd_params
-- PX4 config: px4_sitl_default
-- PX4 platform: posix
-- PX4 lockstep: enabled
-- The CXX compiler identification is AppleClang 17.0.0.17000603
-- The C compiler identification is AppleClang 17.0.0.17000603
-- The ASM compiler identification is AppleClang
-- Found assembler: /usr/bin/cc
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- cmake build type: RelWithDebInfo
-- ccache enabled (export CCACHE_DISABLE=1 to disable)
CMake Deprecation Warning at platforms/posix/src/px4/common/lockstep_scheduler/CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.10 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.


-- Looking for gz-transport13 -- found version 13.5.0
-- Searching for dependencies of gz-transport13
-- Found ZLIB: /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/lib/libz.tbd (found version "1.2.12")
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
-- Config-file not installed for ZeroMQ -- checking for pkg-config
-- Checking for module 'libzmq >= 4'
--   Found libzmq, version 4.3.5
-- Found ZeroMQ: TRUE (Required is at least version "4")
-- Found UUID: TRUE
-- Looking for gz-utils2 -- found version 2.2.1
-- Searching for dependencies of gz-utils2
-- Searching for <gz-utils2> component [cli]
-- Looking for gz-utils2-cli -- found version 2.2.1
-- Searching for dependencies of gz-utils2-cli
-- Looking for gz-msgs10 -- found version 10.3.2
-- Searching for dependencies of gz-msgs10
-- Looking for gz-utils2 -- found version 2.2.1
-- Looking for gz-math7 -- found version 7.5.2
-- Searching for dependencies of gz-math7
-- Looking for gz-utils2 -- found version 2.2.1
-- Checking for module 'tinyxml2'
--   Found tinyxml2, version 11.0.0
-- Found Python3: /Users/ermarinellone/Code/PX4-Autopilot/.venv/bin/python3.14 (found version "3.14.3") found components: Interpreter
-- Found Protobuf: /usr/local/lib/libprotobuf.dylib (found version "6.33.4")
-- Looking for gz-transport13 -- found version 13.5.0
-- Searching for dependencies of gz-transport13
-- Config-file not installed for ZeroMQ -- checking for pkg-config
-- Checking for module 'libzmq >= 4'
--   Found libzmq, version 4.3.5
-- Looking for gz-utils2 -- found version 2.2.1
-- Searching for dependencies of gz-utils2
-- Searching for <gz-utils2> component [cli]
-- Looking for gz-utils2-cli -- found version 2.2.1
-- Searching for dependencies of gz-utils2-cli
-- Looking for gz-msgs10 -- found version 10.3.2
-- Searching for dependencies of gz-msgs10
-- Looking for gz-utils2 -- found version 2.2.1
-- Looking for gz-math7 -- found version 7.5.2
-- Searching for dependencies of gz-math7
-- Looking for gz-utils2 -- found version 2.2.1
-- Checking for module 'tinyxml2'
--   Found tinyxml2, version 11.0.0
-- Looking for gz-sim8 -- found version 8.10.0
-- Searching for dependencies of gz-sim8
-- Looking for sdformat14 -- found version 14.9.0
-- Searching for dependencies of sdformat14
-- Looking for gz-math7 -- found version 7.5.2
-- Looking for gz-utils2 -- found version 2.2.1
-- Looking for gz-plugin2 -- found version 2.0.4
-- Searching for dependencies of gz-plugin2
-- Looking for gz-utils2 -- found version 2.2.1
-- Searching for <gz-plugin2> component [loader]
-- Looking for gz-plugin2-loader -- found version 2.0.4
-- Searching for dependencies of gz-plugin2-loader
-- Searching for <gz-plugin2> component [register]
-- Looking for gz-plugin2-register -- found version 2.0.4
-- Searching for dependencies of gz-plugin2-register
-- Looking for gz-transport13 -- found version 13.5.0
-- Searching for dependencies of gz-transport13
-- Config-file not installed for ZeroMQ -- checking for pkg-config
-- Checking for module 'libzmq >= 4'
--   Found libzmq, version 4.3.5
-- Looking for gz-utils2 -- found version 2.2.1
-- Looking for gz-msgs10 -- found version 10.3.2
-- Searching for <gz-transport13> component [log]
-- Looking for gz-transport13-log -- found version 13.5.0
-- Searching for dependencies of gz-transport13-log
-- Searching for <gz-transport13> component [parameters]
-- Looking for gz-transport13-parameters -- found version 13.5.0
-- Searching for dependencies of gz-transport13-parameters
-- Looking for gz-msgs10 -- found version 10.3.2
-- Looking for gz-common5 -- found version 5.8.0
-- Searching for dependencies of gz-common5
-- Looking for gz-utils2 -- found version 2.2.1
-- Looking for dlfcn.h - found
-- Looking for libdl - found
-- Found DL: TRUE
-- Searching for <gz-common5> component [av]
-- Looking for gz-common5-av -- found version 5.8.0
-- Searching for dependencies of gz-common5-av
-- Checking for module 'libswscale'
--   Found libswscale, version 9.1.100
-- Found SWSCALE: TRUE
-- Checking for module 'libavdevice >= 56.4.100'
--   Found libavdevice, version 62.1.100
-- Found AVDEVICE: TRUE (Required is at least version "56.4.100")
-- Checking for module 'libavformat'
--   Found libavformat, version 62.3.100
-- Found AVFORMAT: TRUE
-- Checking for module 'libavcodec'
--   Found libavcodec, version 62.11.100
-- Found AVCODEC: TRUE
-- Checking for module 'libavutil'
--   Found libavutil, version 60.8.100
-- Found AVUTIL: TRUE
-- Searching for <gz-common5> component [events]
-- Looking for gz-common5-events -- found version 5.8.0
-- Searching for dependencies of gz-common5-events
-- Looking for gz-math7 -- found version 7.5.2
-- Searching for <gz-common5> component [graphics]
-- Looking for gz-common5-graphics -- found version 5.8.0
-- Searching for dependencies of gz-common5-graphics
-- Looking for gz-math7 -- found version 7.5.2
-- Found GzAssimp: TRUE (found version "6.0.4")
-- Searching for <gz-common5> component [io]
-- Looking for gz-common5-io -- found version 5.8.0
-- Searching for dependencies of gz-common5-io
-- Searching for <gz-common5> component [profiler]
-- Looking for gz-common5-profiler -- found version 5.8.0
-- Searching for dependencies of gz-common5-profiler
-- Looking for gz-fuel_tools9 -- found version 9.1.1
-- Searching for dependencies of gz-fuel_tools9
-- Found CURL: /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/lib/libcurl.tbd (found version "8.7.1")
-- Checking for module 'jsoncpp'
--   Found jsoncpp, version 1.9.6
-- Found JSONCPP: TRUE
-- Checking for module 'yaml-0.1'
--   Found yaml-0.1, version 0.2.5
-- Found YAML: TRUE
-- Checking for module 'libzip'
--   Found libzip, version 1.11.4
-- Found ZIP: TRUE
-- Looking for gz-utils2 -- found version 2.2.1
-- Looking for gz-common5 -- found version 5.8.0
-- Searching for dependencies of gz-common5
-- Looking for gz-utils2 -- found version 2.2.1
-- Looking for dlfcn.h - found
-- Looking for libdl - found
-- Searching for <gz-common5> component [testing]
-- Looking for gz-common5-testing -- found version 5.8.0
-- Searching for dependencies of gz-common5-testing
-- Looking for gz-math7 -- found version 7.5.2
-- Looking for gz-msgs10 -- found version 10.3.2
-- Looking for gz-gui8 -- found version 8.4.0
-- Searching for dependencies of gz-gui8
-- Looking for gz-utils2 -- found version 2.2.1
-- Looking for gz-math7 -- found version 7.5.2
-- Looking for gz-common5 -- found version 5.8.0
-- Looking for gz-plugin2 -- found version 2.0.4
-- Looking for gz-transport13 -- found version 13.5.0
-- Looking for gz-rendering8 -- found version 8.2.3
-- Searching for dependencies of gz-rendering8
-- Looking for gz-math7 -- found version 7.5.2
-- Searching for dependencies of gz-math7
-- Looking for gz-utils2 -- found version 2.2.1
-- Searching for <gz-math7> component [eigen3]
-- Looking for gz-math7-eigen3 -- found version 7.5.2
-- Searching for dependencies of gz-math7-eigen3
-- Checking for module 'eigen3'
--   Found eigen3, version 5.0.1
-- Looking for gz-common5 -- found version 5.8.0
-- Searching for dependencies of gz-common5
-- Looking for gz-utils2 -- found version 2.2.1
-- Looking for dlfcn.h - found
-- Looking for libdl - found
-- Searching for <gz-common5> component [graphics]
-- Searching for <gz-common5> component [events]
-- Searching for <gz-common5> component [geospatial]
-- Looking for gz-common5-geospatial -- found version 5.8.0
-- Searching for dependencies of gz-common5-geospatial
-- Looking for gz-math7 -- found version 7.5.2
-- Looking for gz-common5 -- found version 5.8.0
-- Looking for gz-plugin2 -- found version 2.0.4
-- Searching for dependencies of gz-plugin2
-- Looking for gz-utils2 -- found version 2.2.1
-- Searching for <gz-plugin2> component [all]
-- Looking for all libraries of gz-plugin2 -- found version 2.0.4
-- Looking for gz-plugin2 -- found version 2.0.4
-- Looking for gz-utils2 -- found version 2.2.1
-- Looking for gz-msgs10 -- found version 10.3.2
-- Looking for gz-physics7 -- found version 7.6.0
-- Searching for dependencies of gz-physics7
-- Looking for gz-common5 -- found version 5.8.0
-- Looking for gz-math7 -- found version 7.5.2
-- Looking for gz-plugin2 -- found version 2.0.4
-- Looking for gz-utils2 -- found version 2.2.1
-- Searching for <gz-physics7> component [heightmap]
-- Looking for gz-physics7-heightmap -- found version 7.6.0
-- Searching for dependencies of gz-physics7-heightmap
-- Looking for gz-common5 -- found version 5.8.0
-- Searching for <gz-physics7> component [mesh]
-- Looking for gz-physics7-mesh -- found version 7.6.0
-- Searching for dependencies of gz-physics7-mesh
-- Looking for gz-common5 -- found version 5.8.0
-- Searching for <gz-physics7> component [sdf]
-- Looking for gz-physics7-sdf -- found version 7.6.0
-- Searching for dependencies of gz-physics7-sdf
-- Looking for sdformat14 -- found version 14.9.0
-- Looking for gz-sensors8 -- found version 8.2.2
-- Searching for dependencies of gz-sensors8
-- Looking for gz-math7 -- found version 7.5.2
-- Looking for gz-common5 -- found version 5.8.0
-- Looking for gz-transport13 -- found version 13.5.0
-- Looking for gz-rendering8 -- found version 8.2.3
-- Looking for gz-msgs10 -- found version 10.3.2
-- Looking for sdformat14 -- found version 14.9.0
-- Searching for <gz-sensors8> component [air_pressure]
-- Looking for gz-sensors8-air_pressure -- found version 8.2.2
-- Searching for dependencies of gz-sensors8-air_pressure
-- Searching for <gz-sensors8> component [air_speed]
-- Looking for gz-sensors8-air_speed -- found version 8.2.2
-- Searching for dependencies of gz-sensors8-air_speed
-- Searching for <gz-sensors8> component [altimeter]
-- Looking for gz-sensors8-altimeter -- found version 8.2.2
-- Searching for dependencies of gz-sensors8-altimeter
-- Searching for <gz-sensors8> component [imu]
-- Looking for gz-sensors8-imu -- found version 8.2.2
-- Searching for dependencies of gz-sensors8-imu
-- Searching for <gz-sensors8> component [force_torque]
-- Looking for gz-sensors8-force_torque -- found version 8.2.2
-- Searching for dependencies of gz-sensors8-force_torque
-- Searching for <gz-sensors8> component [logical_camera]
-- Looking for gz-sensors8-logical_camera -- found version 8.2.2
-- Searching for dependencies of gz-sensors8-logical_camera
-- Searching for <gz-sensors8> component [magnetometer]
-- Looking for gz-sensors8-magnetometer -- found version 8.2.2
-- Searching for dependencies of gz-sensors8-magnetometer
-- Searching for <gz-sensors8> component [navsat]
-- Looking for gz-sensors8-navsat -- found version 8.2.2
-- Searching for dependencies of gz-sensors8-navsat
-- Searching for <gz-sensors8> component [dvl]
-- Looking for gz-sensors8-dvl -- found version 8.2.2
-- Searching for dependencies of gz-sensors8-dvl
-- Looking for gz-sensors8 -- found version 8.2.2
-- Searching for dependencies of gz-sensors8
-- Looking for gz-math7 -- found version 7.5.2
-- Looking for gz-common5 -- found version 5.8.0
-- Looking for gz-transport13 -- found version 13.5.0
-- Looking for gz-rendering8 -- found version 8.2.3
-- Looking for gz-msgs10 -- found version 10.3.2
-- Looking for sdformat14 -- found version 14.9.0
-- Searching for <gz-sensors8> component [rendering]
-- Looking for gz-sensors8-rendering -- found version 8.2.2
-- Searching for dependencies of gz-sensors8-rendering
-- Searching for <gz-sensors8> component [rendering]
-- Searching for <gz-sensors8> component [lidar]
-- Looking for gz-sensors8-lidar -- found version 8.2.2
-- Searching for dependencies of gz-sensors8-lidar
-- Looking for gz-sensors8 -- found version 8.2.2
-- Searching for <gz-sensors8> component [gpu_lidar]
-- Looking for gz-sensors8-gpu_lidar -- found version 8.2.2
-- Searching for dependencies of gz-sensors8-gpu_lidar
-- Looking for gz-sensors8 -- found version 8.2.2
-- Searching for <gz-sensors8> component [camera]
-- Looking for gz-sensors8-camera -- found version 8.2.2
-- Searching for dependencies of gz-sensors8-camera
-- Looking for gz-sensors8 -- found version 8.2.2
-- Searching for <gz-sensors8> component [boundingbox_camera]
-- Looking for gz-sensors8-boundingbox_camera -- found version 8.2.2
-- Searching for dependencies of gz-sensors8-boundingbox_camera
-- Searching for <gz-sensors8> component [segmentation_camera]
-- Looking for gz-sensors8-segmentation_camera -- found version 8.2.2
-- Searching for dependencies of gz-sensors8-segmentation_camera
-- Searching for <gz-sensors8> component [depth_camera]
-- Looking for gz-sensors8-depth_camera -- found version 8.2.2
-- Searching for dependencies of gz-sensors8-depth_camera
-- Looking for gz-sensors8 -- found version 8.2.2
-- Searching for <gz-sensors8> component [rgbd_camera]
-- Looking for gz-sensors8-rgbd_camera -- found version 8.2.2
-- Searching for dependencies of gz-sensors8-rgbd_camera
-- Looking for gz-sensors8 -- found version 8.2.2
-- Searching for <gz-sensors8> component [thermal_camera]
-- Looking for gz-sensors8-thermal_camera -- found version 8.2.2
-- Searching for dependencies of gz-sensors8-thermal_camera
-- Looking for gz-sensors8 -- found version 8.2.2
-- Searching for <gz-sensors8> component [wide_angle_camera]
-- Looking for gz-sensors8-wide_angle_camera -- found version 8.2.2
-- Searching for dependencies of gz-sensors8-wide_angle_camera
-- Looking for gz-rendering8 -- found version 8.2.3
-- Looking for gz-math7 -- found version 7.5.2
-- Looking for gz-utils2 -- found version 2.2.1
-- Looking for gz-sensors8 -- found version 8.2.2
-- Looking for gz-plugin2 -- found version 2.0.4
-- Found OpenCV: /usr/local/Cellar/opencv/4.13.0_3 (found version "4.13.0")
-- Found OpenCV: /usr/local/Cellar/opencv/4.13.0_3 (found version "4.13.0") found components: core imgproc
-- Found PkgConfig: /usr/local/bin/pkg-config (found version "2.5.1")
-- Checking for module 'gstreamer-1.0'
--   Found gstreamer-1.0, version 1.28.0
-- Checking for module 'gstreamer-app-1.0'
--   Found gstreamer-app-1.0, version 1.28.0
-- Found Java: /Library/Java/JavaVirtualMachines/temurin-25.jdk/Contents/Home/bin/java (found version "25.0.2")
-- ROMFS: ROMFS/px4fmu_common
-- Configuring done (36.9s)
-- Generating done (1.2s)
-- Build files have been written to: /Users/ermarinellone/Code/PX4-Autopilot/build/px4_sitl_default
[0/1162] git submodule src/modules/uxrce_dds_client/Micro-XRCE-DDS-Client
[13/1162] git submodule src/modules/mavlink/mavlink
[17/1162] git submodule src/drivers/gps/devices
[38/1162] Performing download step (git clone) for 'OpticalFlow'
Cloning into 'OpticalFlow'...
Already on 'master'
Your branch is up to date with 'origin/master'.
Submodule 'external/klt_feature_tracker' (https://github.com/PX4/klt_feature_tracker.git) registered for path 'external/klt_feature_tracker'
Cloning into '/Users/ermarinellone/Code/PX4-Autopilot/build/px4_sitl_default/OpticalFlow/src/OpticalFlow/external/klt_feature_tracker'...
Submodule path 'external/klt_feature_tracker': checked out 'a2733b54f17b571dd958be498b57479a4104d8cd'
[80/1162] Building CXX object src/modules/simulation/gz_msgs/CMakeFiles/px4_gz_msgs.dir/opticalflow.pb.cc.o
FAILED: [code=1] src/modules/simulation/gz_msgs/CMakeFiles/px4_gz_msgs.dir/opticalflow.pb.cc.o
/usr/local/bin/ccache /usr/bin/c++ -DCONFIG_ARCH_BOARD_PX4_SITL -DENABLE_LOCKSTEP_SCHEDULER -D__CUSTOM_FILE_IO__ -D__PX4_DARWIN -D__PX4_POSIX -D__STDC_FORMAT_MACROS -Dnoreturn_function="__attribute__((noreturn))" -I/Users/ermarinellone/Code/PX4-Autopilot/boards/px4/sitl/src -I/Users/ermarinellone/Code/PX4-Autopilot/platforms/posix/src/px4/common/include -I/Users/ermarinellone/Code/PX4-Autopilot/build/px4_sitl_default -I/Users/ermarinellone/Code/PX4-Autopilot/build/px4_sitl_default/src/lib -I/Users/ermarinellone/Code/PX4-Autopilot/platforms/posix/src/px4/generic/generic/include -I/Users/ermarinellone/Code/PX4-Autopilot/platforms/common -I/Users/ermarinellone/Code/PX4-Autopilot/platforms/common/include -I/Users/ermarinellone/Code/PX4-Autopilot/src -I/Users/ermarinellone/Code/PX4-Autopilot/src/include -I/Users/ermarinellone/Code/PX4-Autopilot/src/lib -I/Users/ermarinellone/Code/PX4-Autopilot/src/lib/matrix -I/Users/ermarinellone/Code/PX4-Autopilot/src/modules -I/Users/ermarinellone/Code/PX4-Autopilot/platforms/posix/include -I/Users/ermarinellone/Code/PX4-Autopilot/build/px4_sitl_default/external/Install/include -I/Users/ermarinellone/Code/PX4-Autopilot/build/px4_sitl_default/src/modules/simulation/gz_msgs -O2 -g -DNDEBUG -std=gnu++17 -arch x86_64 -fPIC -g -fdata-sections -ffunction-sections -fomit-frame-pointer -fmerge-all-constants -fno-signed-zeros -fno-trapping-math -freciprocal-math -fno-math-errno -fno-strict-aliasing -fvisibility=hidden -include visibility.h -Wall -Wextra -Werror -Warray-bounds -Wcast-align -Wdisabled-optimization -Wdouble-promotion -Wfatal-errors -Wfloat-equal -Wformat-security -Winit-self -Wlogical-op -Wpointer-arith -Wshadow -Wuninitialized -Wunknown-pragmas -Wunused-variable -Wno-missing-field-initializers -Wno-missing-include-dirs -Wno-unused-parameter -fcolor-diagnostics -fdiagnostics-absolute-paths -Qunused-arguments -Wno-c99-designator -Wno-unknown-warning-option -Wno-unused-const-variable -Wno-varargs -Wno-vla-cxx-extension -Wreorder -Wno-overloaded-virtual -MD -MT src/modules/simulation/gz_msgs/CMakeFiles/px4_gz_msgs.dir/opticalflow.pb.cc.o -MF src/modules/simulation/gz_msgs/CMakeFiles/px4_gz_msgs.dir/opticalflow.pb.cc.o.d -o src/modules/simulation/gz_msgs/CMakeFiles/px4_gz_msgs.dir/opticalflow.pb.cc.o -c /Users/ermarinellone/Code/PX4-Autopilot/build/px4_sitl_default/src/modules/simulation/gz_msgs/opticalflow.pb.cc
In file included from /Users/ermarinellone/Code/PX4-Autopilot/build/px4_sitl_default/src/modules/simulation/gz_msgs/opticalflow.pb.cc:6:
In file included from /Users/ermarinellone/Code/PX4-Autopilot/build/px4_sitl_default/src/modules/simulation/gz_msgs/opticalflow.pb.h:26:
In file included from /usr/local/Cellar/protobuf/33.4_1/include/google/protobuf/generated_message_reflection.h:26:
In file included from /usr/local/Cellar/protobuf/33.4_1/include/google/protobuf/descriptor.h:56:
In file included from /usr/local/Cellar/protobuf/33.4_1/include/google/protobuf/extension_set.h:36:
In file included from /usr/local/Cellar/abseil/20260107.1/include/absl/container/btree_map.h:66:
In file included from /usr/local/Cellar/abseil/20260107.1/include/absl/container/internal/btree.h:68:
In file included from /usr/local/Cellar/abseil/20260107.1/include/absl/container/internal/layout.h:198:
/usr/local/Cellar/abseil/20260107.1/include/absl/strings/str_cat.h:353:61: fatal error: implicit conversion increases floating-point precision: 'float' to 'double' [-Wdouble-promotion]
  353 |       : piece_(digits_, numbers_internal::SixDigitsToBuffer(f, digits_)) {}
      |                         ~~~~~~~~~~~~~~~~                    ^
1 error generated.
[91/1162] Building CXX object src/modules/simulation/gz_plugins/buoyancy/CMakeFiles/BuoyancySystemPlugin.dir/BuoyancySystem.cpp.o
ninja: build stopped: subcommand failed.
make: *** [px4_sitl] Error 1
```

Platform: MacOS

Build succeeds adding the compile option ignoring the double promotion in _src/modules/simulation/gz_msgs/CMakeLists.txt_.
```
    target_compile_options(px4_gz_msgs PRIVATE -Wno-error=double-promotion)
```

### Flight Log / Additional Information

_No response_

--- a/src/modules/simulation/gz_msgs/CMakeLists.txt
+++ b/src/modules/simulation/gz_msgs/CMakeLists.txt
@@ -18,6 +18,9 @@ if (Protobuf_FOUND)
         ${PROTO_HDRS}
     )
 
+    # Disable double-promotion warning for protobuf-generated code
+    target_compile_options(px4_gz_msgs PRIVATE -Wno-double-promotion)
+
     target_include_directories(px4_gz_msgs
         PUBLIC
             ${CMAKE_CURRENT_BINARY_DIR}

